File: Flutter/rtc_conference_tui_kit/lib/conference/conference_observer.dart
import 'package:rtc_conference_tui_kit/conference/conference_error.dart';

class ConferenceObserver {
  Function(String conferenceId, ConferenceError error)? onConferenceStarted;
  Function(String conferenceId, ConferenceError error)? onConferenceJoined;
  Function(String conferenceId)? onConferenceFinished;
  Function(String conferenceId)? onConferenceExited;

  ConferenceObserver(
      {this.onConferenceStarted,
      this.onConferenceJoined,
      this.onConferenceFinished,
      this.onConferenceExited});
}

File: iOS/TUIRoomKit/Source/ViewModel/ConferenceObserver.swift
//
//  ConferenceObserver.swift
//  TUIRoomKit
//
//  Created by 唐佳宁 on 2024/3/15.
//

import Foundation

@objc public protocol ConferenceObserver {
    @objc optional func onConferenceStarted(conferenceId: String, error: ConferenceError)
    @objc optional func onConferenceJoined(conferenceId: String, error: ConferenceError)
    @objc optional func onConferenceFinished(conferenceId: String)
    @objc optional func onConferenceExited(conferenceId: String)
}
